### PR TITLE
Add editorconfig style checker CI.

### PR DESCRIPTION
**Pull request by quyykk**
## Feature Details

This PR adds a CI check for editorconfig violations using https://github.com/editorconfig-checker/editorconfig-checker.

It also fixes every style check found by the tool.

## Testing Done

Tested the CI on my branch: https://github.com/quyykk/endless-sky/pull/5

### CODE REVIEW

**Comment by quyykk** [comment]
What's the reason for not trimming trailing whitespace at the end of data files?

**Comment by tehhowch** [comment]
> What's the reason for not trimming trailing whitespace at the end of data files?

History. So those without IDEs or smart text editors can "jump right in."

I happily conceded that anyone modifying the C++ code these days is equipped with an editor capable of understanding the indentation syntax needed, but I don't agree that is the case for those who wish to modify the data files.

I don't have a strong opinion on continuing to enforce trailing whitespace on _new_ content. I see existing content as falling in the "not broken, don't fix it" category.

**Comment by quyykk** [comment]
Fair enough. There was an error in the pattern matching inside the the editorconfig file which I fixed. However, I had to "fix" deliberate uses of spaces for alignment. Not ideal but if we use/recommend editorconfig our code shouldn't have violations, even if they're false positives.

File: source/FillShader.cpp [hunk 1]
```diff
@@ -72,9 +72,9 @@ void FillShader::Init()
 
 	GLfloat vertexData[] = {
 		-.5f, -.5f,
-		 .5f, -.5f,
+		.5f, -.5f,
 		-.5f,  .5f,
-		 .5f,  .5f
+		.5f,  .5f
```
Comment (Amazinite) [inline source/FillShader.cpp:77]: Still have some double spaces here.

File: .editorconfig [hunk 2]
```diff
@@ -30,3 +35,8 @@ indent_size = 2
 [*.{yml,yaml}]
 indent_size = 2
 indent_style = space
+
+# Ignore assets
+[{images,sounds}/**]
+insert_final_newline = unset
+trim_trailing_whitespace = unset
```
Comment (tehhowch) [inline .editorconfig:42]: Why do we need to exclude these directories?

**Review by tehhowch — COMMENTED** [review]
Seems like we may need to configure the checker a bit

File: .editorconfig (hunk 2 above)
Comment (quyykk) [reply .editorconfig:42]: It's because the checker seems to think some asset files are text files and emits a ton of errors about trailing whitespace.

File: .editorconfig [hunk 3]
```diff
@@ -9,10 +9,15 @@ trim_trailing_whitespace = true
 [data/**.txt]
 trim_trailing_whitespace = false
 indent_style = tab
+[.codespell.exclude]
+trim_trailing_whitespace = false
```
[the remark this answers is not in the document]
Comment (quyykk) [reply .editorconfig:13]: `trim_trailing_whitespace` is for whitespace preceding newlines. As such, it doesn't differentiate between empty lines with whitespace and whitespace at the end of non-empty lines. I can't find another option too, so it's either both or nothing. Editorconfigs seems really inflexible when it comes to options.

[the remark this answers is not in the document]
Comment (tehhowch) [reply .editorconfig:13]: Let me rephrase - why do we need to exclude the _spellcheck file_ from trailing whitespace checks?

I posit that the only reason there should be trailing whitespace in the spellcheck file would be due to trailing whitespace on a line with a would-be typo. But, we don't want trailing whitespace on a `data/` line _with content_, so if there exist any such lines, they should be fixed.

File: .editorconfig (hunk 2 above)
Comment (tehhowch) [reply .editorconfig:42]: Looks like .wav and .mp3 aren't in the [default excludes](https://github.com/editorconfig-checker/editorconfig-checker#default-excludes), so we would need to add them.

PNG are definitely in the excluded list, though, so not sure which files in `images/` are causing the issue.

File: .editorconfig (hunk 3 above)
[the remark this answers is not in the document]
Comment (tehhowch) [reply .editorconfig:13]: Addressed existing trailing whitespace on content lines in https://github.com/endless-sky/endless-sky/commit/e76614fc8abf65de07f3ce20cb26eb070f63696b, so you do not need to treat the codespell file specially

File: .editorconfig (hunk 2 above)
Comment (quyykk) [reply .editorconfig:42]: Are those excludes official? Because if not some other editorconfig checker tool could try to verify images as well (but I don't know). It's seems safer to simpler ignore everything in those two folders.
